### ISSUE TITLE
Terraform keeps destroy/recreate kusto cluster without any change on it

### ISSUE THREAD

**Issue by ekersale** [hashicorp/terraform-provider-azurerm#14307]
Hi,

We are facing an issue with terraform azurerm provider on azurerm_kusto_cluster. 

Each time we are running terraform it detects breaking change on the resource that trigger destruction and recreation even when there is absolutely no change made on it. 

The plan proposed by terraform is the following: 

```
-/+ resource "azurerm_kusto_cluster" "kusto" {
      ~ data_ingestion_uri        = "https://ingest-xxxxxx.canadacentral.kusto.windows.net" -> (known after apply)
      - enable_streaming_ingest   = false -> null
      ~ id                        = "/subscriptions/xxxxx-xxxxx-xxxxx-xxxx-xxxxxxxxxxxx/resourceGroups/psp-prod01-kusto/providers/Microsoft.Kusto/Clusters/xxxx" -> (known after apply)
      - language_extensions       = [] -> null
        name                      = "xxxx"
      - tags                      = {} -> null
      ~ trusted_external_tenants  = [
          - "*",
        ] -> (known after apply)
      ~ uri                       = "https://xxxxxxx.canadacentral.kusto.windows.net" -> (known after apply)
      ~ zones                     = [ # forces replacement
            "1",
          - "3",
            "2",
          + "3",
        ]
        # (6 unchanged attributes hidden)

      ~ identity {
          - identity_ids = [] -> null
          ~ principal_id = "xxxxxxx-xxxxxx-xxxxx-xxxxx-xxxxxxxxxxxx" -> (known after apply)
          ~ tenant_id    = "***" -> (known after apply)
            # (1 unchanged attribute hidden)
        }


        # (2 unchanged blocks hidden)
    }
```

However we are not making any change on the zones.

We are running terraform v1.0.5 and azure rm provider v2.81.0

**Comment by ms-henglu** [comment]
Hi @ekersale ,

Thanks for reporting this issue, I have a workaround to fix it, would you please give it a try?
Workaround 1. Change zones to ["1", "3", "2"]
Workaround 2. Add the following config to azurerm_kusto_cluster resource
```
 lifecycle {
    ignore_changes = [
      zones,
    ]
  }
```

And I'll raise a PR to change zones to Set, it will suppress order difference.

**Comment by ekersale** [comment]
Hi @ms-henglu,

Thanks for the workaround this is not perfect but at least it does fix the issue temporarily!

**Comment by ms-henglu** [comment]
Hi @ekersale ,

This change has been merged, but it's a 3.0 feature, here's how to opt into 3.0: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-beta

refs: https://github.com/hashicorp/terraform-provider-azurerm/pull/14342#issuecomment-1044181515

**Comment by magodo** [comment]
Closing this issue as it is fixed in one of the linked PR in issue: https://github.com/hashicorp/terraform-provider-azurerm/issues/14588.
